Modules/_localemodule.c

```diff
@@ -87,14 +89,19 @@ PyLocale_setlocale(PyObject* self, PyObject* args)
     int category;
     char *locale = NULL, *result;
     PyObject *result_object;
+    _locale_state *state;
 
     if (!PyArg_ParseTuple(args, "i|z:setlocale", &category, &locale))
         return NULL;
 
 #if defined(MS_WINDOWS)
     if (category < LC_MIN || category > LC_MAX)
     {
-        PyErr_SetString(Error, "invalid locale category");
+        state = PyModule_GetState(self);
+        if (state == NULL) {
+            return NULL;
+        }
```

Comment:
I don't think that it's worth it to check for error: this function cannot fail, self is always a module object.

I would prefer a function like:

```
static inline _locale_state* get_locale_state(PyObject *mod)
{
    void *state = PyModule_GetState(mod);
    assert(state != NULL);
    return (_locale_state*)state;
}
```

And then use:

```
PyErr_SetString(get_locale_state(self)->Error, "invalid locale category"); 
```

Reply:
Make sense, I update this PR.

Do we need add a global macros of `get_state`?
I found some other extension modules use local macro do this behavior too. such as: https://github.com/python/cpython/blob/master/Modules/_hashopenssl.c#L55

Thanks a million, victor. You helpe me merge much PRs today ;)

Reply:
A `static inline` function is a bit better than a macro.

Reply:
Sure, I remember victor have explained in other bpo/PR(oh, I forgot which one is). It's easy for debuging and something else.
I will try to add this `inline function` in free time.